File: autoit-mediapipe-com/src/binding/tasks/vision/pose_landmarker.cc
#include "binding/tasks/vision/pose_landmarker.h"

namespace mediapipe::tasks::vision::pose_landmarker {
	using Connection = PoseLandmarksConnections::Connection;

	const std::vector<Connection>& PoseLandmarksConnections::POSE_LANDMARKS() {
		static const std::vector<Connection> POSE_LANDMARKS = {
			{ 0, 1 },
			{ 1, 2 },
			{ 2, 3 },
			{ 3, 7 },
			{ 0, 4 },
			{ 4, 5 },
			{ 5, 6 },
			{ 6, 8 },
			{ 9, 10 },
			{ 11, 12 },
			{ 11, 13 },
			{ 13, 15 },
			{ 15, 17 },
			{ 15, 19 },
			{ 15, 21 },
			{ 17, 19 },
			{ 12, 14 },
			{ 14, 16 },
			{ 16, 18 },
			{ 16, 20 },
			{ 16, 22 },
			{ 18, 20 },
			{ 11, 23 },
			{ 12, 24 },
			{ 23, 24 },
			{ 23, 25 },
			{ 24, 26 },
			{ 25, 27 },
			{ 26, 28 },
			{ 27, 29 },
			{ 28, 30 },
			{ 29, 31 },
			{ 30, 32 },
			{ 27, 31 },
			{ 28, 32 },
		};

		return POSE_LANDMARKS;
	}
}

PTR_BRIDGE_IMPL(mediapipe::tasks::vision::pose_landmarker::PoseLandmarkerResultRawCallback);

template<typename _In, typename _Out>
inline const HRESULT autoit_to_callback(VARIANT const* in_val, _Out& result_callback) {
	_In c_callback;
	HRESULT hr = autoit_to(in_val, c_callback);
	if (SUCCEEDED(hr)) {
		result_callback = ::mediapipe::tasks::autoit::core::utils::CppConvertToResultCallback(c_callback);
	}
	return hr;
}

const HRESULT autoit_to(VARIANT const* in_val, mediapipe::tasks::vision::pose_landmarker::PoseLandmarkerResultCallback& result_callback) {
	using _In = mediapipe::tasks::vision::pose_landmarker::PoseLandmarkerResultRawCallback;
	using _Out = mediapipe::tasks::vision::pose_landmarker::PoseLandmarkerResultCallback;
	return autoit_to_callback<_In, _Out>(in_val, result_callback);
}

File: src/declarations/tasks/vision/pose_landmarker/pose_landmarker.js
const ns = "mediapipe::tasks::vision::pose_landmarker";
const nsProgId = ns.replaceAll("::", ".");
const fqn = `${ ns }::PoseLandmarker`;
const RunningMode = "tasks::vision::core::RunningMode";

module.exports = ({language}) => {
    const progid = `mediapipe.tasks.${ language }.vision.pose_landmarker`;
    const utils = `mediapipe::tasks::${ language }::core::utils`;

    return [
        // expose pose_landmarker properties like in mediapipe python
        [`${ progid }.`, "", ["/Properties"], [
            [`${ ns }::PoseLandmarkerOptions`, "PoseLandmarkerOptions", "", ["/R", "=this", "/S"]],
            [`${ ns }::PoseLandmarker`, "PoseLandmarker", "", ["/R", "=this", "/S"]],
        ], "", ""],

        [`struct ${ nsProgId }.PoseLandmarkerOptions`, "", [`/progid=${ progid }.PoseLandmarkerOptions`, "/Simple", "/DC"], [
            ["tasks::core::BaseOptions", "base_options", "", ["/RW"]],
            ["core::RunningMode", "running_mode", `${ RunningMode }::IMAGE`, ["/RW"]],
            ["int", "num_poses", "1", ["/RW"]],
            ["float", "min_pose_detection_confidence", "0.5f", ["/RW"]],
            ["float", "min_pose_presence_confidence", "0.5f", ["/RW"]],
            ["float", "min_tracking_confidence", "0.5f", ["/RW"]],
            ["bool", "output_segmentation_masks", "false", ["/RW"]],
            ["PoseLandmarkerResultCallback", "result_callback", "", [`/WExpr=${ utils }::CppConvertToResultCallback($value, $0, $hr)`]],
        ], "", ""],

        [`class ${ nsProgId }.PoseLandmarker`, "", [`/progid=${ progid }.PoseLandmarker`], [
            ["mediapipe::tasks::components::containers::Connection", "Connection", "", ["/R", "=this", "/S"]],
        ], "", ""],

        [`${ nsProgId }.PoseLandmarker.Create`, `std::shared_ptr<${ fqn }>`, ["=create_from_model_path", "/S", `/WrapAs=${ utils }::ReleaseToSharedPtr`], [
            ["std::string", "model_path", "", [`/Expr=std::unique_ptr<PoseLandmarkerOptions>(new PoseLandmarkerOptions{ .base_options{ .model_asset_path{ $0 } }, .running_mode{ ${ RunningMode }::IMAGE } })`]],
        ], "", ""],

        [`${ nsProgId }.PoseLandmarker.Create`, `std::shared_ptr<${ fqn }>`, ["=create_from_options", "/S", `/WrapAs=${ utils }::ReleaseToSharedPtr`], [
            ["std::shared_ptr<PoseLandmarkerOptions>", "options", "", ["/Expr=tasks::autoit::core::utils::ConvertToUniquePtr($0)"]],
        ], "", ""],

        [`${ nsProgId }.PoseLandmarker.Detect`, "absl::StatusOr<PoseLandmarkerResult>", ["=detect"], [
            ["Image", "image", "", []],
            ["std::optional<core::ImageProcessingOptions>", "image_processing_options", "std::nullopt", []],
        ], "", ""],

        [`${ nsProgId }.PoseLandmarker.DetectForVideo`, "absl::StatusOr<PoseLandmarkerResult>", ["=detect_for_video"], [
            ["Image", "image", "", []],
            ["uint64_t", "timestamp_ms", "", []],
            ["std::optional<core::ImageProcessingOptions>", "image_processing_options", "std::nullopt", []],
        ], "", ""],

        [`${ nsProgId }.PoseLandmarker.DetectAsync`, "absl::Status", ["=detect_async"], [
            ["Image", "image", "", []],
            ["uint64_t", "timestamp_ms", "", []],
            ["std::optional<core::ImageProcessingOptions>", "image_processing_options", "std::nullopt", []],
        ], "", ""],

        [`${ nsProgId }.PoseLandmarker.Close`, "absl::Status", ["=close"], [], "", ""],
    ];
};
